src/_cffi_src/openssl/callbacks.py

```diff
@@ -50,4 +59,81 @@
     # backwards compatibility for old cffi version on PyPy
     # and Python >=3.5 (https://github.com/pyca/cryptography/issues/2970)
     TYPES = "static const long Cryptography_STATIC_CALLBACKS;"
-    CUSTOMIZATIONS = "static const long Cryptography_STATIC_CALLBACKS = 0;"
+    CUSTOMIZATIONS = """static const long Cryptography_STATIC_CALLBACKS = 0;
+"""
+
+
+if sysconfig.get_config_var("WITH_THREAD"):
+    FUNCTIONS += """
```

Comment:
I think you need an additional guard for OpenSSL < 1.1.0 and not LibreSSL. OpenSSL 1.1.0 has its own locking code.